### PR TITLE
Fix send money show "$0.00" in preview card

### PR DESCRIPTION
**Pull request by mollfpr**
@luacmartins @mountiny 

### Details

In the report actions list, we use the action data from `originalMessage` to provide the **_amount_**, **_currency_**, and **_comment_** to pass it to `IOUPreview`.

For request money and request money paid, the `originalMessage` has structured `{..., amount: ..., currency: ..., comment: ...}` but for the send money, the object is put in `IOUDetails`, so it will be `{..., IOUDetails: {amount: ..., currency: ..., comment: ...}}`. In this case, the component `IOUPreview` failed to get the data because it's not expecting it to be in the `IOUDetails`.

As it's confirmed by @mountiny [here](https://github.com/Expensify/App/issues/18881#issuecomment-1546742811) that it's the correct structure, we need to make a new utils method to return the `{amount: ..., currency: ..., comment: ...}` with checking the `IOUDetails` key if it's provided.

### Fixed Issues

$ https://github.com/Expensify/App/issues/18881

### Tests

**Prerequisite**
1. Provide two accounts in the different browsers or platform
    - **Account A** will be the main account to test, and the ones who request money and get paid
    - **Account B** will be the secondary account that receive the requested money and pays the **Account A**

**Send Money**
1. Open the App with **Account A**
2. Go to the global actions button (The big + green button) and select **Send money**
3. Put any amount
4. Search the **Account B**
5. Put any description and submit the sent money
6. **Verify that the IOU preview displays the correct amount, currency, and comment with a green checkmark as it's settled**

Screenshot 2023-05-14 at 12 43 49

**Request Money**
1. Open the App with **Account A**
2. Go to the global actions button (The big + green button) and select **Request money**
3. Put any amount
4. Search the **Account B**
5. Put any description and submit the request for money
6. **Verify that the IOU preview displays the correct amount, currency, and comment**
8. Open the App with **Account B**
9. Find the chat from **Account A**
10. **Verify that the IOU preview displays the correct amount, currency, and comment that Account A sent**
11. Press the **Pay** button and press button **I'll settle up anywhere** to settle the requested money
12. **Verify that the IOU preview still displays the correct amount and currency with a green checkmark as it's settled**
13. Open the App with **Account A** and open the chat with **Account B**
14. **Verify that the IOU preview display new action with the correct amount and currency with a green checkmark as it's settled**

Screenshot 2023-05-14 at 12 42 24

- [x] Verify that no errors appear in the JS console

### Offline tests

**Send Money Offline**
1. Open the App with **Account A**
2. Go to the global actions button (The big + green button) and select **Send money**
3. Put any amount
4. Search the **Account B**
5. Put any description and submit the sent money
6. **Verify that the IOU preview is darker and displays the correct amount, currency, and comment with a green checkmark as it's settled**

**Request Money Offline**
1. Open the App with **Account A**
2. Go to the global actions button (The big + green button) and select **Request money**
3. Put any amount
4. Search the **Account B**
5. Put any description and submit the request for money
6. **Verify that the IOU preview displays the correct amount, currency, and comment**
8. Open the App with **Account B**
9. Find the chat from **Account A**
10. **Verify that the IOU preview is darker and displays the correct amount, currency, and comment that Account A sent**

**Requested Money from Account A**
1. Open the App with **Account B** and find the chat with **Account A**
2. Press the **Pay** button in the IOU preview and press the button **I'll settle up anywhere** to settle the requested money
3. **Verify that the IOU preview is darker and still displays the correct amount and currency with a green checkmark as it's settled**

### QA Steps

Same as the Tests

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

https://github.com/Expensify/App/assets/25520267/a2807512-4fc8-40df-a3d5-0eb17076f146

</details>

<details>
<summary>Mobile Web - Chrome</summary>

https://github.com/Expensify/App/assets/25520267/8515be0a-ba54-45c0-b01e-26854f8fa79c

</details>

<details>
<summary>Mobile Web - Safari</summary>

https://github.com/Expensify/App/assets/25520267/87c603c8-0de6-47c3-b5a9-4ead48097ec2

</details>

<details>
<summary>Desktop</summary>

https://github.com/Expensify/App/assets/25520267/c243f55e-4b77-45a8-b782-9666b1b1154a

</details>

<details>
<summary>iOS</summary>

https://github.com/Expensify/App/assets/25520267/bdf42bf6-d0db-4694-9c96-fa405936bb6a

</details>

<details>
<summary>Android</summary>

https://github.com/Expensify/App/assets/25520267/9298ab3a-6d8b-4e92-b375-ba0e445a8bc3

</details>

### CODE REVIEW

File: src/libs/ReportUtils.js [hunk 1]
```diff
@@ -937,6 +937,32 @@ function getDisplayNamesWithTooltips(participants, isMultipleParticipantReport)
...
 
+/**
+ * We get the amount, currency and comment money request value from the action.originalMessage.
+ * But for the send money action, the above value is put in the IOUDetails object.
+ *
+ * @param {Object} reportAction
+ * @param {Number} reportAction.amount
+ * @param {String} reportAction.currency
+ * @param {String} reportAction.comment
+ * @param {Object} reportAction.IOUDetails
```
Comment (mountiny) [inline src/libs/ReportUtils.js:948]: to make sure its clear its not there every time
```suggestion
 * @param {Object} [reportAction.IOUDetails]
```

**Review by mountiny — DISMISSED** [review]
@mollfpr looking good, thanks for working on this over the weekend 🙇  When will this be ready for review?

**Review by luacmartins — DISMISSED** [review]

**Comment by mollfpr** [comment]
@luacmartins @mountiny Fix the conflict and address the commented JSDoc.

**Review by luacmartins — APPROVED** [review]

**Review by luacmartins — APPROVED** [review]

**Review by luacmartins — DISMISSED** [review]

**Comment by mountiny** [comment]
We get it @luacmartins

**Comment by mountiny** [comment]
@mollfpr we just merged a PR to main which as a merge conflict not, can you please resolve the conflict now, thank you!

**Comment by luacmartins** [comment]
lol Github was being slow so I spammed the button. oops

**Comment by mollfpr** [comment]
Handle it now

**Review by mountiny — APPROVED** [review]
Thanks @mollfpr

**Review by luacmartins — APPROVED** [review]

**Comment by mountiny** [comment]
@mollfpr Can you merge in main again, I think that there were some small fixes in the flow

I tested a bit and seems to work fine.

@sobitneupane will you be able to go through the testing on Monday?

**Comment by mollfpr** [comment]
Merged current main

**Comment by sobitneupane** [comment]
> @sobitneupane will you be able to go through the testing on Monday?

Yup. I will test it today.

**Review by MonilBhavsar — APPROVED** [review]

**Comment by sobitneupane** [comment]
@mollfpr I am not seeing IOUPreview at all on `Send Money`. Can you please test it in your end?
Screenshot 2023-05-15 at 13 20 57

**Comment by mollfpr** [comment]
@sobitneupane Yeah, it's not showing in me either. It's due to changes in here https://github.com/Expensify/App/pull/18898

@mountiny Could you confirm if we intended not to show the IOUPreview on the settled money request?

**Comment by luacmartins** [comment]
@sobitneupane @mollfpr yes that's intentional for deleted and pay messages

**Comment by mollfpr** [comment]
@luacmartins So the issue isn’t valid anymore? 🥲

**Comment by luacmartins** [comment]
@mollfpr  Yea, I think we don't need this anymore

**Comment by mollfpr** [comment]
What do you think @mountiny? I think it's okay to close this.

**Comment by mountiny** [comment]
image

the created previews will change design but we wont add a new preview for settled up

I actually cant find a mocks of Send money flow so I am not sure cc @trjExpensify

**Comment by mountiny** [comment]
Looking that the send money will still add the `create` action and then pays it immediately so the preview should show there

**Review by Julesssss — APPROVED** [review]

**Comment by mountiny** [comment]
## Reviewer Checklist

- [ ] I have verified the author checklist is complete (all boxes are checked off).
- [ ] I verified the correct issue is linked in the `### Fixed Issues` section above
- [ ] I verified testing steps are clear and they cover the changes made in this PR
    - [ ] I verified the steps for local testing are in the `Tests` section
    - [ ] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [ ] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [ ] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [ ] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android / native
    - [ ] Android / Chrome
    - [ ] iOS / native
    - [ ] iOS / Safari
    - [ ] MacOS / Chrome / Safari
    - [ ] MacOS / Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [ ] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [ ] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [ ] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [ ] I verified that comments were added to code that is not self explanatory
    - [ ] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [ ] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [ ] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [ ] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [ ] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [ ] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [ ] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [ ] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [ ] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [ ] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [ ] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [ ] If a new component is created I verified that:
    - [ ] A similar component doesn't exist in the codebase
    - [ ] All props are defined accurately and each prop has a `/** comment above it */`
    - [ ] The file is named correctly
    - [ ] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [ ] The only data being stored in the state is data necessary for rendering and nothing else
    - [ ] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [ ] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [ ] All JSX used for rendering exists in the render method
    - [ ] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [ ] If any new file was added I verified that:
    - [ ] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [ ] If a new CSS style is added I verified that:
    - [ ] A similar style doesn't already exist
    - [ ] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG`)
- [ ] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [ ] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [ ] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [ ] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>



</details>

<details>
<summary>Mobile Web - Chrome</summary>



</details>

<details>
<summary>Mobile Web - Safari</summary>



</details>

<details>
<summary>Desktop</summary>



</details>

<details>
<summary>iOS</summary>



</details>

<details>
<summary>Android</summary>



</details>

**Comment by mountiny** [comment]
Shipping this I need it for another PR

File: src/components/ReportActionItem/IOUPreview.js [hunk 2]
```diff
@@ -144,9 +143,12 @@ const IOUPreview = (props) => {
...
     const isCurrentUserManager = managerEmail === sessionEmail;
 
+    const moneyRequestAction = ReportUtils.getMoneyRequestAction(props.action);
+
     // If props.action is undefined then we are displaying within IOUDetailsModal and should use the full report amount
-    const requestAmount = props.isIOUAction ? lodashGet(props.action, 'originalMessage.amount', 0) : ReportUtils.getMoneyRequestTotal(props.iouReport);
-    const requestCurrency = props.isIOUAction ? lodashGet(props.action, 'originalMessage.currency', CONST.CURRENCY.USD) : props.iouReport.currency;
+    const requestAmount = props.isIOUAction ? moneyRequestAction.total : ReportUtils.getMoneyRequestTotal(props.iouReport);
+    const requestCurrency = props.isIOUAction ? moneyRequestAction.currency : props.iouReport.currency;
+    const requestComment = Str.htmlDecode(moneyRequestAction.comment).trim()
```
Comment (mountiny) [inline src/components/ReportActionItem/IOUPreview.js:151]: Fuuu why did the prettier not catch this 😡

Comment (marcaaron) [reply src/components/ReportActionItem/IOUPreview.js:151]: idk but it is creating some major weirdness for me locally. Is anyone fixing it?

Comment (mountiny) [reply src/components/ReportActionItem/IOUPreview.js:151]: https://github.com/Expensify/App/pull/18985

**Comment by mountiny** [comment]
This wasnt CPed because of the linter issue so removed the label to make it clear

**Pull request merged**